cli/cmd/compose/images.go

```diff
@@ -0,0 +1,103 @@
+/*
+   Copyright 2020 Docker Compose CLI authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package compose
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/docker/compose-cli/api/client"
+	"github.com/docker/compose-cli/cli/formatter"
+	"github.com/docker/compose-cli/utils"
+	"github.com/docker/docker/pkg/stringid"
+
+	units "github.com/docker/go-units"
+)
+
+type imageOptions struct {
+	*projectOptions
+	Quiet bool
+}
+
+func imagesCommand(p *projectOptions) *cobra.Command {
+	opts := imageOptions{
+		projectOptions: p,
+	}
+	imgCmd := &cobra.Command{
+		Use:   "images [SERVICE...]",
+		Short: "List images used by the created containers",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runImages(cmd.Context(), opts, args)
+		},
+	}
+	imgCmd.Flags().BoolVarP(&opts.Quiet, "quiet", "q", false, "Only display IDs")
+	return imgCmd
+}
+
+func runImages(ctx context.Context, opts imageOptions, services []string) error {
+	c, err := client.New(ctx)
+	if err != nil {
+		return err
+	}
+
+	project, err := opts.toProject(services)
```

Comment:
better use `toProjectName`, as backend don't need an actual project, just name

Reply:
In the backend we need the sevice names from the project to filter the container list. I could pass the project name and the service list in the api method if you think it would be better.

Reply:
In the backend we need the sevice names from the project to filter the container list. I could pass the project name and the service list in the api method if you think it would be better.

Reply:
indeed, see https://github.com/docker/compose-cli/pull/1509/files#diff-bf7596d83ffc37f2610a3a7b826911de05f2c7996bdf3f6ad4add0a88de35195R29-R30